### PR TITLE
[481287] Fixed bug in plugin.xml content comparison of xtext generator

### PR DESCRIPTION
**Pull request by cdietrich**
[481287] Fixed bug in plugin.xml content comparison of xtext generator
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=481287

**Comment by cdietrich** [comment]
please review

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGenerator.xtend [hunk 1]
```diff
@@ -338,7 +338,7 @@ class XtextGenerator extends AbstractWorkflowComponent2 {
 			if (root.isFile(pluginXml.path)) { 
 				// only write plugin.xml_gen if entries exist and content differs
 				if (!pluginXml.entries.isEmpty
-					&& root.readTextFile(pluginXml.path) != pluginXml.getContent 
+					&& root.readTextFile(pluginXml.path)?.toString != pluginXml.getContent?.toString
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGenerator.xtend]: Should we introduce TextFileAccess#getContentString to easier avoid the same confusion / bug pattern in the future?

Comment (cdietrich) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGenerator.xtend]: the problem is the other side of the comparison. text file access already does

```
	def CharSequence getContent() {
		return internalContents.toString
	}
```

Comment (cdietrich) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGenerator.xtend]: => at least i wont need to second toString

Comment (cdietrich) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGenerator.xtend]: well the code is overriden for some stange reason

Comment (cdietrich) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGenerator.xtend]: updated the code

File: org.eclipse.xtext.xtext.generator/xtend-gen/org/eclipse/xtext/xtext/generator/XtextGenerator.java [hunk 2]
```diff
@@ -447,7 +447,29 @@ protected void generatePluginXmls() {
         final IXtextGeneratorFileSystemAccess root = entry.getValue();
         boolean _isFile = root.isFile(pluginXml.getPath());
         if (_isFile) {
-          if ((((!pluginXml.getEntries().isEmpty()) && (!Objects.equal(root.readTextFile(pluginXml.getPath()), pluginXml.getContent()))) && pluginXml.getPath().endsWith(".xml"))) {
+          boolean _and = false;
```
Comment (kthoms) [inline org.eclipse.xtext.xtext.generator/xtend-gen/org/eclipse/xtext/xtext/generator/XtextGenerator.java]: I'm wondering why the code is verbose again. With 2.11 Xtend should produce the less verbose code.

Comment (cdietrich) [inline org.eclipse.xtext.xtext.generator/xtend-gen/org/eclipse/xtext/xtext/generator/XtextGenerator.java]: i dont know

**Comment by spoenemann** [comment]
👍

**Pull request merged**